src/synapse.py

```diff
@@ -83,6 +91,47 @@ def execute_migrate_config(self, container: ops.Container) -> None:
                 "Migrate config failed, please review your charm configuration"
             )
 
+    def server_name_configured(self, container: ops.Container) -> str | None:
+        """Get server_name from configuration file.
+
+        Args:
+            container: Container of the charm.
+
+        Returns:
+            str | None: server_name or None if configuration file is not found.
+        """
+        try:
+            configuration_content = str(
+                container.pull(SYNAPSE_CONFIG_PATH, encoding="utf-8").read()
+            )
+        except PathError:
+            logger.error("configuration file %s does not exist", SYNAPSE_CONFIG_PATH)
+            return None
+        return yaml.safe_load(configuration_content)["server_name"]
+
+    def reset_instance_action(self, container: ops.Container) -> None:
```

Comment:
What is the reason for including the `_action` postfix in this function please? Just thinking that this is coupled with the charm.py in terms of the name?

Reply:
I followed the example here: https://juju.is/docs/sdk/actions "_on_grant_admin_role_action"
But I don't mind changing it. Is there a different standard/guideline for this?

Reply:
That standard is more applicable to `charm.py`, within the synapse module it isn't applicable. The applicable standard is the guidance in the managing charm complexity doc around that modules shouldn't be charm aware. Does that make sense?

Reply:
Oh sorry, I didn't notice. Changed!